src/boot/efi/meson.build

```diff
@@ -62,6 +62,8 @@ if have_gnu_efi
         efi_conf = configuration_data()
         efi_conf.set_quoted('PACKAGE_VERSION', meson.project_version())
         efi_conf.set_quoted('EFI_MACHINE_TYPE_NAME', EFI_MACHINE_TYPE_NAME)
+        efi_conf.set('SD_BOOT_LOG_TPM', get_option('tpm'))
```

Comment:
I'm not sure what you mean, if you use `mesonconf -Dtpm=false` in `#undef SD_BOOT_LOG_TPM` is written to `efi_conf.h` and for `true` `#define SD_BOOT_LOG_TPM` is written.

Isn't that what `#ifdef` checks? I tested both configuration options and the seem to work.